app/webapp/src/main/java/uk/ac/ebi/fg/annotare2/web/gwt/editor/client/view/experiment/design/DataAssignmentViewImpl.java

```diff
@@ -176,28 +177,36 @@ private String getDataFileColumnName(FileType type, int index) {
 
     private List<EnumWithHelpText> getAllowedColumnTypes() {
         List<EnumWithHelpText> types = new ArrayList<>();
-        for (FileType type : FileType.values()) {
-            if(!type.isProcessedMatrix()){
-                if (experimentType.isSequencing()){
-                    types.add(type);
-                }
-                else if (isMicroArrayOrTwoColourMicroArrayWithNonRawMatrixFileType(type)){
-                    alwaysAddProcessedTypeToListAndOnlyAddOtherTypesOnce(type, types);
-                }
-            }
+        if(experimentType.isSequencing()){
+            addSequencingAllowedTypes(types);
+        } else if(experimentType.isTwoColorMicroarray()) {
+            addTwoColorMicroarrayAllowedTypes(types);
+        } else if(experimentType.isMicroarray()) {
+            addOneColorMicroarrayAllowedTypes(types);
         }
         return types;
     }
 
-    private boolean isMicroArrayOrTwoColourMicroArrayWithNonRawMatrixFileType(FileType type) {
-        return experimentType.isMicroarray() ||
-                (experimentType.isTwoColorMicroarray() && !type.isRawMatrix());
+    private static void addTypesToList(List<EnumWithHelpText> types, FileType... fileTypes) {
```

Comment:
All of the above methods are using side effects. Are these actions really necessary?
Are there better ways to refactor these methods?
Thanks